src/ranges.jl

```diff
@@ -193,7 +203,7 @@ specified then indices for visiting each index of `x` is returned.
 """
 @inline function indices(x)
   inds = eachindex(x)
-  if inds isa AbstractUnitRange{<:Integer}
+  if inds isa AbstractUnitRange#{<:Integer} # prevents inference
```

Comment:
Could maybe use `inds isa AbstractUnitRange && eltype(inds) <: Integer`?

Reply:
Done, and inference tests pass.